apps/assessment_registry/dashboard_schema.py

```diff
@@ -0,0 +1,335 @@
+import graphene
+from dataclasses import dataclass
+from collections import defaultdict
+
+
+from django.db.models import Count
+from django.db import models
+from django.contrib.postgres.aggregates.general import ArrayAgg
+from django.db.models.functions import TruncDay
+from django.db import connection as django_db_connection
+
+from deep.caches import CacheHelper
+from geo.models import GeoArea
+from utils.graphene.enums import EnumDescription
+from .enums import (
+    AssessmentRegistryAffectedGroupTypeEnum,
+    AssessmentRegistryCoordinationTypeEnum,
+    AssessmentRegistryDataCollectionTechniqueTypeEnum,
+    AssessmentRegistryFocusTypeEnum,
+    AssessmentRegistryProtectionInfoTypeEnum,
+    AssessmentRegistrySectorTypeEnum,
+)
+from deep_explore.schema import count_by_date_queryset_generator
+from .filter_set import (
+    AssessmentDashboardFilterDataInputType,
+    AssessmentDashboardFilterSet,
+)
+from .models import AssessmentRegistry, AssessmentRegistryOrganization, MethodologyAttribute
+
+
+def get_global_filters(_filter: dict, date_field="created_at"):
+    return {
+        f"{date_field}__gte": _filter["date_from"],
+        f"{date_field}__lte": _filter["date_to"],
+    }
+
+
+@dataclass
+class AssessmentDashboardStat:
+    cache_key: str
+    assessment_registry_qs: models.QuerySet
+    methodology_attribute_qs: models.QuerySet
+
+
+class AssessmentDashboardFilterInputType(graphene.InputObjectType):
+    date_from = graphene.Date(required=True)
+    date_to = graphene.Date(required=True)
+    assessment = AssessmentDashboardFilterDataInputType()
```

Comment:
I think we need to handle error when the filter is not provided.

Reply:
@thenav56 , the above gives error when filter not provided . Do we need to haldle error for this ? Or It will be provided by default in client side.

Reply:
@Rup-Narayan-Rajbanshi  filter is optional field